homework/list.js

```diff
@@ -1,79 +1,142 @@
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
+(function (exports) {
+    /**
+     *  Declare _listNoteContent.
+     *  Declare wrapper.
+     *
+     *  @constructor
+     *  @this {TodoListManager}
+     */
+    var TodoListManager = function () {
+        // Local data storage; should sync up with the server.
+        this._listNoteContent = [];
+        // Will be an element as the list wrapper.
+        this._wrapper = null;
     };
-  }
-
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    }
-  }
-
-  function updateList(list) {
-    _listNoteContent = list;
-  }
-
-  function drawList() {
-    var list = _listNoteContent;
-    var ul = document.createElement('ul');
-    ul.id = 'note-title-list';
-    var buff = document.createDocumentFragment();
-    list.forEach(function(note, i) {
-      var li = document.createElement('li');
-      li.dataset.noteId = i;
-      li.classList.add('note-title');
-      li.textContent = note.title;
-      // Note: buff is captured, so we now have a
-      // little closure naturally.
-      buff.appendChild(li);
-    });
-    ul.appendChild(buff);
-    _wrapper.appendChild(ul);
-  }
-
-  function fetchList(afterFetch) {
-    var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
-    xhr.responseType = 'json';
-    xhr.onreadystatechange = function(e) {
-      // Watch out: we have a mysterious unknown 'this'.
-      if (this.readyState === 4 && this.status === 200) {
-        var listData = this.response;
-        // The flow ends here.
-        afterFetch(listData);
-      } else if (this.status !== 200 ){
-        // Ignore error in this case.
-      }
+    TodoListManager.prototype = {
+        /**
+         * Initial wrapper
+         * Add a eventlistener to listen click event 
+         * Call fetchList to fetech data,
+         * after fetch , draw the list
+         * 
+         * @this {TodoListManager}
+         */
+        start() {
+            this._wrapper = document.querySelector('#note-list-wrapper');
+            window.addEventListener('click', this);
+            this.fetchList().then((function (data) {
+                this.updateList(data);
+                this.drawList();
+                this.preloadFirstNote();
+            }).bind(this))
+        },
+        /**
+         * Dispatch a custom event when mouse click at the note-title.
+         * Also determine the content to be showed.
+         *
+         * @param {Object} event- mouse click event.
+         * @this {TodoListManager}
+         */
+        onNoteOpen(event) {
+            if (event.target.classList.contains('note-title')) {
```

Comment:
If a function body is wrapped by an `if...else` with only one branch, moving the condition into the caller rather than caller is recommended.